book/internals/wrap_gtk_widget.nim

```diff
@@ -0,0 +1,201 @@
+  - Use the nim equivalent for any C-type in the docs: `cbool` for `bool`, `cfloat` for `float`, `cdouble` for `double` etc.
+  - If the docs of a function mention that the caller needs to free the memory of the return value, then use a managed type for the return-type such as `OwnedGtkString` for cstrings (see e.g. `g_file_get_path`)
+
+For examples of constructor functions, look for procs with the `_new` suffix in [GTK.nim](https://github.com/can-lehmann/owlkettle/blob/main/owlkettle/gtk.nim).
+
+### *Add Initialization to the Widget*
+Next we need to tell Owlkettle create the GTK widget during the construction of the Owlkettle widget.
+This is done as part of Owlkettle's `beforeBuild` hook.
+
+Add a `beforeBuild` hook to the widget type in `widgets.nim`.
+All it needs to do is call the constructor for your widget and assign the created `GtkWidget` to `state.internalWidget`.
+If the constructor requires parameters, add fields with their values to your Widget and access the values from the implicitly available `widget` variable and its `val/has<Fieldname>` fields. 
+
+For examples, see the [beforeBuild docs](https://can-lehmann.github.io/owlkettle/book/internals/hooks/before_build_hook.html) or search for `beforeBuild:` in [widgets.nim](https://github.com/can-lehmann/owlkettle/blob/main/owlkettle/widgets.nim).
+
+### *See your example in action!*
+Once you're at this step you should be able to compile your example and see your widget in action!
+Run it with `nim r --path:. ./examples/path/to/your/example.nim` from the base dir of your Owlkettle repository clone.
+
+Use your example to manually test your widget as you add features to it in the later section.
+
+Remember to add signal-handler procs with bodies to the example, once you have added signal event listeners.
+That way you can see your application react more to various inputs.
+
+For a short example, look at Owlkettle's [counter-example](https://github.com/can-lehmann/owlkettle/blob/main/examples/counter.nim).
+
+
+### *Add fields to the Widget*
+Beyond just creating the widget and providing event-listeners, GTK may provide options to further customize a widget.
+
+E.g. you may be able to show or hide values, change their positioning, enable or disable them etc. by updating certain fields at runtime after the widget was constructed.
+
+Take a look at the functions of your widget (and the functions of its parent classes) in the GTK docs, specifically anything that isn't a getter.
+If there's procs to add or set a property of the widget, try to add that feature to your wrapped widget.
+If the proc requires a parameter of type `GtkWidget` or an array/list of `GtkWidget`, add a field of type `Widget`/`seq[Widget]` to your wrapped widget instead of `GtkWidget`. 
+
+At minimum, try to wrap all features exposed by your widget directly and possibly its direct parent (e.g. for `Gtk.Scale` everything from there as well as its parent `Gtk.Range`)
```

Comment:
```suggestion
Take a look at the properties of the GTK widget.
At minimum, try to wrap all properties exposed by your widget and possibly its direct parent (e.g. for `Gtk.Scale` everything from there as well as its parent `Gtk.Range`)
```

adders are not really explained here

Reply:
Good point.

Reply:
This section should not discuss how to add adders though, it should definitely only be a link as you suggested.